pegasus/rake/generate_pdfs.rake

```diff
@@ -18,23 +19,57 @@ def pdf_conversions_for_files(file_pattern, url_extension)
   end
 end
 
+def pdf_conversions_for_state_pages(state_codes)
+  [].tap do |state_conversion_infos|
+    state_codes.each do |state_code|
+      file_path_without_makepdf_extension = sites_v3_dir('code.org/public/advocacy/state-facts/') + state_code
+      file_path_without_extension = file_path_without_makepdf_extension
+      url_path_from_public_without_extension = file_path_without_extension.match(/public(.*)/)[1]
+      url_path = url_path_from_public_without_extension + '?pdf_version=true'
+      state_conversion_infos << PDFConversionInfo.new(url_path, [file_path_without_makepdf_extension], file_path_without_extension + '.pdf')
+    end
+  end
+end
+
 base_url = ENV['base_url']
 
+def generate_pdf_files(base_url, pdf_conversion_info, fetchfile_for_pdf)
+  url = "#{base_url}#{pdf_conversion_info.url_path}"
+
+  PDF.generate_from_url(url, pdf_conversion_info.output_pdf_path, verbose: true)
+
+  fetchable_url = RakeUtils.replace_file_with_s3_backed_fetch_file(pdf_conversion_info.output_pdf_path, fetchfile_for_pdf, bucket: 'cdo-fetch')
+
+  HipChat.log "Created <b>#{pdf_conversion_info.output_pdf_path}</b> and moved to <a href='#{fetchable_url}'>#{fetchable_url}</a></b>."
+end
+
 all_outfiles = [].tap do |all_outfiles|
+  # Generate pdf for files that are appended with .makepdf
   pdf_conversions_for_files(sites_v3_dir('code.org/**/[^_]*.makepdf'), '').each do |pdf_conversion_info|
     fetchfile_for_pdf = "#{pdf_conversion_info.output_pdf_path}.fetch"
 
     file fetchfile_for_pdf => pdf_conversion_info.src_files do
-      url = "#{base_url}#{pdf_conversion_info.url_path}"
-
-      PDF.generate_from_url(url, pdf_conversion_info.output_pdf_path, verbose: true)
+      generate_pdf_files(base_url, pdf_conversion_info, fetchfile_for_pdf)
+    end
 
-      fetchable_url = RakeUtils.replace_file_with_s3_backed_fetch_file(pdf_conversion_info.output_pdf_path, fetchfile_for_pdf, bucket: 'cdo-fetch')
+    all_outfiles << fetchfile_for_pdf
+  end
 
-      HipChat.log "Created <b>#{pdf_conversion_info.output_pdf_path}</b> and moved to <a href='#{fetchable_url}'>#{fetchable_url}</a></b>."
+  # Generate pdf for each state using code.org/public/advocacy/state-facts/splat.haml
+  [].tap do |states|
+    DB[:cdo_state_promote].each do |state_row|
+      states << state_row[:state_code_s]
     end
 
-    all_outfiles << fetchfile_for_pdf
+    pdf_conversions_for_state_pages(states).each do |pdf_info|
+      fetchfile_for_pdf = "#{pdf_info.output_pdf_path}.fetch"
+
+      file fetchfile_for_pdf do
```

Comment:
you don't use the src_files dependencies here (like how it's used above for the .makepdf files. Is this not necessary here? If not, is it necessary above?)